File: leetcode-04/solutions/cpp/0034.cpp
class Solution {
 public:
  vector<int> searchRange(vector<int>& nums, int target) {
    int leftIndex = find(nums, target, true);

    if (leftIndex == nums.size() || nums[leftIndex] != target) return {-1, -1};
    return {leftIndex, find(nums, target, false) - 1};
  }

 private:
  int find(vector<int>& nums, int target, bool isLeft) {
    int l = 0;
    int r = nums.size();

    while (l < r) {
      int m = (l + r) >> 1;
      if (target < nums[m] || (isLeft && target == nums[m]))
        r = m;
      else
        l = m + 1;
    }

    return l;
  }
};

File: leetcode-04/solutions/python3/0034.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        leftIndex = self.find(nums, target, True)

        if leftIndex == len(nums) or nums[leftIndex] != target:
            return -1, -1
        return leftIndex, self.find(nums, target, False) - 1

    def find(self, nums: List[int], target: int, isLeft: bool) -> int:
        l = 0
        r = len(nums)

        while l < r:
            m = (l + r) >> 1
            if target < nums[m] or (isLeft and target == nums[m]):
                r = m
            else:
                l = m + 1

        return l
